codex/utils/keyutils.nim

```diff
@@ -26,18 +28,17 @@ proc setupKey*(path: string): ?!PrivateKey =
   if not path.fileAccessible({AccessFlags.Find}):
     info "Creating a private key and saving it"
     let
-      res = ? PrivateKey.random(Rng.instance()[]).mapFailure(CodexKeyError)
-      bytes = ? res.getBytes().mapFailure(CodexKeyError)
+      res = ? PrivateKey.random(Rng.instance()[]).mapFailure()
```

Comment:
This explicitly maps key related errors (which is an enum iirc), into a `CodexKeyError`, what the exception type it's being mapped to now?

Reply:
I was having problems getting compilation to work with `mapFailure(CodexKeyError)` so I tried making the error types more generic. Part of it was my misunderstanding about the Result types and generic type conversions on  them, and part of it changes to compiler.

But I figured out the issue with `mapFailure` and should be able to change this back.